draft-ietf-quic-load-balancers.md

```diff
@@ -449,66 +449,52 @@ The nonce length MUST be no fewer than 4 octets.
 The configuration agent also selects an 16-octet AES-ECB key to use for
 connection ID decryption.
 
-### Load Balancer Actions {#encrypted-short-load-balancer-actions}
+### Server Actions
 
-Upon receipt of a QUIC packet, the load balancer extracts as many of the
-earliest octets from the destination connection ID as necessary to match the
-server ID. The nonce immediately follows.
+When generating a routable connection ID, the server writes arbitrary bits into
+its nonce octets, and its provided server ID into the server ID octets. See
+{{cid-entropy}} for nonce generation considerations.
 
-The load balancer decrypts the nonce and the server ID using the following three
-pass algorithm:
+The server encrypts the server ID using the following four pass algorithm, which
+leverages 128-bit AES Electronic Codebook (ECB) mode, much like QUIC header
+protection.
 
-* Pass 1:  The load balancer decrypts the server ID using 128-bit AES Electronic
-Codebook (ECB) mode, much like QUIC header protection. The encrypted nonce
-octets are zero-padded to 16 octets.  AES-ECB encrypts this encrypted nonce
-using its key to generate a mask which it applies to the encrypted server id.
-This provides an intermediate value of the server ID, referred to as server-id
-intermediate.
+In the text below, ^ is the XOR function, || is concatenation, and "zeros" is of
+length equal to 15 octets minus the server ID length, so that the input to
+AES-ECB is 16 octets.
 
-server_id_intermediate = encrypted_server_id ^ AES-ECB(key, padded-encrypted-nonce)
+The truncate() function takes the most significant octets of its argument, so that
+the XOR function operates on fields of the same length.
 
-* Pass 2:  The load balancer decrypts the nonce octets using 128-bit AES
-ECB mode, using the server-id intermediate as "nonce" for this pass. The
-server-id intermediate octets are zero-padded to 16 octets.  AES-ECB encrypts
-this padded server-id intermediate using its key to generate a mask which it
-applies to the encrypted nonce. This provides the decrypted nonce value.
+First, set server_id_0 to the server ID, and nonce_0 to the nonce.
 
-nonce = encrypted_nonce ^ AES-ECB(key, padded-server_id_intermediate)
+Then apply the following steps:
 
-* Pass 3:  The load balancer decrypts the server ID using 128-bit AES ECB mode.
-The nonce octets are zero-padded to 16 octets.  AES-ECB encrypts this nonce
-using its key to generate a mask which it applies to the intermediate server id.
-This provides the decrypted server ID.
+    server_id_1 = server_id_0 ^ (truncate(AES-ECB(key, nonce_0 || zeros || 0x01)))
```

Comment:
Ok, good suggestion. I'll try to clarify the language.

Reply:
Actually, looking again, the text already says "zeros is of length equal to 15 octets minus the server ID length, so that the input to AES-ECB is 16 octets."

As for padding, since we've specified a 16B input, I suppose the padding setting is irrelevant. Why have a normative requirement?

That said, and example would be useful, and I'll add one.